File: BinSearch.py
def bSearch(): 
   arr=[]
   n = int(input("Enter number of elements:"))
   print("Enter the elements in the array:")
   for i in range(0, n): 
       num=int(input()) 
       arr.append(num)
   x=int(input("Enter the element to be Searched:"))
   high=len(arr)
   low=0
   mid=0
   while high>=low:
    mid=low+(high-low)//2
    if arr[mid]==x:
      return mid
    elif arr[mid]>x:
      high=mid-1
    else:
      low=mid+1
   return -1

ans=bSearch()
if ans!=-1:
  print("Element is present at position",str(ans+1))
else:
  print("Element is not present in the array")

File: BinSearch.java
//package ques2;
import java.util.*;
 class BinarySearch {


    public static void main(String[] args) {
        Scanner y=new Scanner(System.in);
        System.out.println("Enter the size of array:");
        int n=y.nextInt();
        int arr[]=new int[n];
        System.out.println("Enter the elements of the array in sorted way(increasing order:)");
        for(int i=0;i<n;i++)
        {
            arr[i]=y.nextInt();
        }
        System.out.println("Enter the number to be searched:");
        int x=y.nextInt();
        
        //Arrays.sort(arr); //if the array is not sorted, uncomment this
        int index=bsearch(arr,n-1,0,x);
        if(index==-1)
            System.out.println("Searched number is not present in the array");
        else
            System.out.println("Searched number is present at position "+(index+1)+" in the array");
                
        
        
    }
    
    public static int bsearch(int arr[],int high, int low,int searchedNum)
    {
        while(high>=low)
        {
            int mid=low+(high-low)/2;
            if(arr[mid]==searchedNum)
                return mid;
            if(arr[mid]>searchedNum)
                high=mid-1;
            else
                low=mid+1;
            
        }
        return -1;
     }
    
    
}
      
